src/grid/becke.py

```diff
@@ -162,7 +162,8 @@ def generate_weights(self, points, atcoords, atnums, *, select=None, pt_ind=None
             [
                 self._radii[num]
                 if not np.isnan(self._radii[num])
-                else self._radii[num - 1]
+                else np.nan_to_num(self._radii[num - 1])
+                or np.nan_to_num(self._radii[num - 2])
```

Comment:
A comment here would be helpful to clarify how the `np.nan` radii are treated. I like the `else` statement; it's pretty clever. It took me a quick try to see what is happening, so not bad to explain that as well.